### PR TITLE
Restore and copy sourceimage

### PR DESCRIPTION
**Pull request by chregu**
* Added `\Rokka\Client\Image::restoreSourceImage($hash, $organization)` for restoring deleted images.
* Added `\Rokka\Client\Image::copySourceImage($hash, $destination, $overwrite, $organization)` for copying an image to another organization. (needs backend merge first)

### CODE REVIEW

File: src/Image.php [hunk 1]
```diff
@@ -132,6 +132,66 @@ public function deleteSourceImage($hash, $organization = '')
...
+        return '200' == $response->getStatusCode();
+    }
+
+    /**
+     * Copy a source image to another org.
+     *
+     * @param string $hash           Hash of the image
+     * @param string $destinationOrg The destination organization
+     * @param bool   $overwrite      If an existing image should be overwritten
+     * @param string $organization   Optional organization name
```
Comment (dbu) [inline src/Image.php]: should this be called $sourceOrg ?

File: src/Image.php [hunk 2]
```diff
@@ -132,6 +132,66 @@ public function deleteSourceImage($hash, $organization = '')
...
+
+            throw $e;
+        }
+
+        return '200' == $response->getStatusCode();
+    }
+
+    /**
+     * Copy a source image to another org.
+     *
```
Comment (dbu) [inline src/Image.php:162]: should this explain the permission requirements?

**Review by dbu — COMMENTED** [review]
nice! just some comments on the phpdoc.

why is this marked as WIP? and can you check the build error?

**Comment by chregu** [comment]
it fails due to some regression introduced by sami. Here's the bug report for that https://github.com/FriendsOfPHP/Sami/issues/335

**Comment by chregu** [comment]
ah, and it's marked WIP, because the backend isn't merged&deployed yet for that feature

File: src/Image.php (hunk 2 above)
Comment (chregu) [reply src/Image.php:162]: added

File: src/Image.php (hunk 1 above)
Comment (chregu) [reply src/Image.php]: changed

**Comment by chregu** [comment]
Backend is deployed and ready. Can be reviewed/merged. The failing test is due to a Sami autogenerated docs error, we can live with that for now (Sami is aware of it -> https://github.com/FriendsOfPHP/Sami/issues/335 )

**Pull request merged**